torch/_dynamo/variables/tensor.py

```diff
@@ -604,6 +627,121 @@ def inline_torch_function_unwrapped(
             return tx.inline_user_function_return(tf_func_var, tf_args, {})
 
 
+class NumpyTensorVariable(TensorVariable):
+    """
+    Represents a numpy.ndarray, but backed by torch Tensor. Use this for Tensor.numpy() call.
+    """
+
+    def __init__(
+        self,
+        proxy: torch.fx.Proxy,
+        dtype=None,
+        device=None,
+        layout=None,
+        ndim=None,
+        size=None,
+        stride=None,
+        requires_grad=None,
+        is_quantized=None,
+        is_contiguous=None,
+        is_sparse=None,
+        class_type=torch.Tensor,
+        specialized_value=None,
+        **kwargs,
+    ):
+        super().__init__(
+            proxy,
+            dtype,
+            device,
+            layout,
+            ndim,
+            size,
+            stride,
+            requires_grad,
+            is_quantized,
+            is_contiguous,
+            is_sparse,
+            class_type,
+            specialized_value,
+            **kwargs,
+        )
+
+    def var_getattr(self, tx, name):
+        from torch_np._dtypes import DType
+
+        from torch._dynamo.variables import GetAttrVariable
+        from .builder import TupleVariable, wrap_fx_proxy_cls
+
+        result = None
+        options = VariableTracker.propagate(self)
+        if name == "shape" and self.size is not None:
+            sizes = [ConstantVariable(x) for x in self.size]
+            result = TupleVariable(sizes)
+        elif name == "ndim" and self.ndim is not None:
+            result = ConstantVariable(self.ndim, **options)
+        elif name == "T":
```

Comment:
Having to manually handle the attribute accesses here seems suboptimal; it would have been better if torch_np handled it. If possible, I'd prefer to land the PR *without* this code.

Reply:
This is handled in `torch_np`, yes. These functions dispatch to the correct `Tensor` functions.

Reply:
Since ```NumpyTensorVariable``` extends from ```TensorVariable```, why don't just call the parent's ```var_getattr```?